heareval/baseline.py

```diff
@@ -121,21 +121,27 @@ def frame_audio(audio: Tensor, frame_size: int, hop_size: int, is_centered: bool
     """
     batch_size, num_samples = audio.shape
 
-    # Adjust the number of samples if centering to allow for a half
-    # frame size number of padded samples padded at the start.
+    # If centering, we are going to pad the start and end of the clip
+    # by frame_size // 2. Here, we increase the number of samples to that
+    # size before calculating the number of frames for this hop size.
     start_pad = 0
     if is_centered:
-        half_frame_size = int(frame_size // 2)
-        num_samples += half_frame_size
-        start_pad = half_frame_size
+        start_pad = int(frame_size // 2)
+        num_samples += start_pad * 2
 
-    # Number of frames is the number of hops that can occur within num_samples
-    num_frames = math.ceil(num_samples / hop_size)
+    # Number of frames that will fully fit within num_samples.
+    num_frames = 1 + (num_samples - frame_size) // hop_size
 
-    # Pad audio to facilitate centered frames.
-    padded_num_samples = (num_frames - 1) * hop_size + frame_size
-    end_pad = padded_num_samples - num_samples
-    audio = F.pad(audio, (start_pad, end_pad))
+    # Number of samples after applying framing.
+    framed_num_samples = (num_frames - 1) * hop_size + frame_size
+
```

Comment:
I don't get it. Why don't we just pad with frame_size // 2 at the end too?

Reply:
Need to add in comment to make this more clear -- but basically padding the end with frame_size // 2 is what we do. And then we get to framing, which truncates the padded clip to make sure that a whole number of frames can be returned. In this code instead of actually padding the input and then truncated -- we make the calculations as if we padded, and then only add the number of zeros we need.